src/factories/createCompilerCallback.js

```diff
@@ -46,12 +46,23 @@ export default (compiler: Compiler, callback: Function): Function => {
 
     debug('requestMap', requestMap);
 
-    const entryChunkName = Object.keys(compiler.options.entry)[0];
+    let entryChunkName = 'main';
+
+    if (!compiler.options.entry.length) {
+      entryChunkName = Object.keys(compiler.options.entry)[0];
+    }
```

Comment:
in my project we don't have named entry point so entryChunkName will incorrectly be set to '0'. This change will add a fall back value for that (as a 'main.js' will be generated). 
The relevant webpack config snippet that works:
```
entry: [
    path.join(process.cwd(), 'app/app.js'),
    'eventsource-polyfill',
    'webpack-hot-middleware/client',
  ],

  // Don't use hashes in dev mode for better performance
  output: {
    filename: '[name].js',
    chunkFilename: '[name].chunk.js',
  },
```